.github/workflows/ci_release.yml

```diff
@@ -104,13 +108,27 @@ jobs:
 
       # Generate the binaries and release
       - uses: goreleaser/goreleaser-action@v6
+        if: github.event_name == 'release'
+        with:
+          distribution: goreleaser
+          version: latest
+          args: release --clean
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          GPG_FINGERPRINT: ${{ steps.import_gpg.outputs.fingerprint }}
+          GORELEASER_CURRENT_TAG: ${{ github.event.release.tag_name }}
```

Comment:
Take the tag from release name. This should solve the problem with bad automatic tag fetch (due to `git` limited knowledge and `goreleaser` as a result).